### PR TITLE
flake8-docstrings

### PR DESCRIPTION
**Pull request by Nickolaim**
Bring checks for flake8-docstrings; fix the code.

### CODE REVIEW

File: src/catkin_pkg/cmake.py [hunk 1]
```diff
@@ -38,33 +38,33 @@
...
+    """
     rel_path = os.path.join('templates', 'metapackage.cmake.in')
     return os.path.join(os.path.dirname(__file__), rel_path)
 
 
 def configure_file(template_file, environment):
-    '''
-    Evaluate a .in template file used in CMake with configure_file().
+    """
+    Evaluate a .in template file used in CMake with configure_string().
```
Comment (dirk-thomas) [inline src/catkin_pkg/cmake.py]: Can you please explain why you change this?

File: .travis.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ python:
 install:
   - pip install nose coverage argparse python-dateutil docutils pyparsing
   - pip install flake8 flake8-blind-except flake8-builtins flake8-class-newline flake8-comprehensions flake8-deprecated
+  - pip install flake8-docstrings
```
Comment (dirk-thomas) [inline .travis.yml]: Please insert this package into the previous line which already installs all flake8 plugins.

Comment (Nickolaim) [reply .travis.yml]: Done.  Please note that I plan to add 2 more packages, and the line will be even longer.

File: src/catkin_pkg/cmake.py (hunk 1 above)
Comment (Nickolaim) [reply src/catkin_pkg/cmake.py]: The original comment triggered `D402 First line should not be the function's "signature"` issue.

The description  does not make sense to me - _`configure_file()` is doing `Evaluate .. with configure_file()`_.

Here is the function body:
```
    with open(template_file, 'r') as f:
        template = f.read()
        return configure_string(template, environment)
```

I assumed that there is a typo in the original comment, and replaced `configure_file()` with `configure_string()`.

The alternative is to revert the change and to suppress D402.

Comment (dirk-thomas) [reply src/catkin_pkg/cmake.py]: This Python function has the same functionality as CMakes `configure_file` function. So the comment should stay as is and the warning can be suppressed.

Comment (Nickolaim) [reply src/catkin_pkg/cmake.py]: Fixed. 
Note - the suppression says 'Ignore only D402'; in reality, all issues in the function definition and the comment are suppressed. I was not able to find a way to suppress the issue for the comment only.

**Comment by Nickolaim** [comment]
Anything else that needs to be addressed in this PR?

File: src/catkin_pkg/cmake.py [hunk 3]
```diff
@@ -38,17 +38,17 @@
...
     :returns: ``str`` location of the metapackage CMakeLists.txt CMake template
-    '''
+    """
     rel_path = os.path.join('templates', 'metapackage.cmake.in')
     return os.path.join(os.path.dirname(__file__), rel_path)
 
 
-def configure_file(template_file, environment):
-    '''
+def configure_file(template_file, environment):  # noqa: ignore=D402
```
Comment (dirk-thomas) [inline src/catkin_pkg/cmake.py]: The reason why this comment suppresses all warnings is the way it is written. Please use `# noqa: D402` to only suppress that specific warning. That style ensures that other warnings are not suppressed.

Comment (Nickolaim) [reply src/catkin_pkg/cmake.py]: Fixed.  

There are some other incorrectly suppressed warnings in this project, I will address them in a separate PR.

**Comment by dirk-thomas** [comment]
Thank you for your effort on improving the style!

**Pull request merged**